src/main/java/com/google/devtools/build/lib/windows/WindowsSubprocess.java

```diff
@@ -140,35 +141,32 @@ public Thread newThread(Runnable runnable) {
     stderrStream =
         stderrRedirected ? null : new ProcessInputStream(WindowsProcesses.getStderr(nativeProcess));
     stdinStream = new ProcessOutputStream();
-    waitLatch = new CountDownLatch(1);
     // Every Windows process we start consumes a thread here. This is suboptimal, but seems to be
     // the sanest way to reconcile WaitForMultipleObjects() and Java-style interruption.
-    @SuppressWarnings("unused")
-    Future<?> possiblyIgnoredError = WAITER_POOL.submit(this::waiterThreadFunc);
+    processFuture = WAITER_POOL.submit(this::waiterThreadFunc);
   }
 
-  private void waiterThreadFunc() {
+  // Waits for the process to finish. Returns 'false' if the process exited
```

Comment:
I find it surprising that `false` means "everything went well" and `true` means "there was an error". Do you agree? If so, could you flip the logic around?

Reply:
I agree! Reversing the logic unfortunately makes the code later uses the result hard to understand, so let's go for the most beautiful solution and use an enum here. :)